### ISSUE TITLE
Figure out some way to trigger ::marker without being a list item

### ISSUE THREAD

**Issue by tabatkins** [w3c/csswg-drafts#6781]
In <https://github.com/w3c/csswg-drafts/issues/6738>, @MatsPalmgren brings up a perf issue they noticed where a document with a ton of `summary` elements becomes extremely slow, because each of them touches the `list-item` counter due to being `display: list-item` (because we're using `::marker` to display the disclosure triangle).

The perf problem is tangential (it applies to any enormous list), but in that issue Mats correctly points out that the fact we're using `display: list-item` at all is pretty hacky. For context, this was decided on [back in 2016/2017](https://github.com/whatwg/html/issues/722), in a conversation spanning implementors, WHATWG editors, and CSSWG editors. It didn't have a *great* motivation behind it other than it seemed to work reasonably well and didn't involve us inventing anything new.

That said, maybe we should invent something new? I do wonder if this is compat-bound by now, tho.

**Comment by MatsPalmgren** [comment]
There's also no reason why the [`list-item`](https://drafts.csswg.org/css-display/#typedef-display-listitem) keyword should be restricted to inline/block layout. A list item should be able to have any `display` type IMO. That's a separate issue, but probably worth addressing at the same time since generalizing `::marker` gives you 95% of the implementation for that.

**Comment by tabatkins** [comment]
I agree, but note we skipped that originally based on the impl complexity of supporting `::marker` on arbitrary display types.

(Tho we already support `::before` and `::after` on literally everything; iirc these do require special handling and it's not just a matter of "adding one more name" to the existing machinery.)

**Comment by MatsPalmgren** [comment]
FYI, I implemented a **working prototype** for that in **2019** in Gecko. At that time, I also implemented `table-caption` and `table-cell` as outer display values so that they could have arbitrary inner layout, like grid and flex. I seem to recall that I commented on a csswg issue at the time that all this was fairly easy to support in Gecko.  It must have been some other vendor that was the source of the "impl complexity" assessment.

Anyway, I'd be happy to dust off those patches if the csswg is interested in moving forward on this topic...

**Comment by fantasai** [comment]
Three possible ways forward here:

- Introduce a new keyword to `display` that acts like `list-item` but doesn't touch the `list-item` counter, e.g. `marker-item` or something.
- Have `::marker { content: anything-except-normal-or-none }` trigger creation of a marker box. (This is slightly complicated by the fact that `::marker { content: normal }` needs to create a marker box when attached to a list item.)
- Address the perf problem by setting `counter-increment` to a value that explicitly doesn't touch the list-item counter.

**Comment by Loirooriol** [comment]
I'm afraid that by now there will be various sites using `::marker { content: "> " }` as a synonym of `list-style-type: "> "`, and generating a marker for non-list-items will be bad.
